test/parallel/test-runner-watch-mode.mjs

```diff
@@ -90,10 +90,12 @@ async function testWatch({
     currentRun = '';
     const fileToRenamePath = tmpdir.resolve(fileToUpdate);
     const newFileNamePath = tmpdir.resolve(`test-renamed-${fileToUpdate}`);
-    const interval = setInterval(() => renameSync(fileToRenamePath, newFileNamePath), common.platformTimeout(1000));
+    const interval = setInterval(() => {
+      renameSync(fileToRenamePath, newFileNamePath);
+      clearInterval(interval);
+    }, common.platformTimeout(1000));
```

Comment:
Hey @cjihrig, regarding these tests, I was working on a GitHub codespace yesterday, and I noticed that this test was flaky there.
The issue was with the interval: if the system is delayed for any reason beyond the timeout, a second run of "rename" (and similarly for "delete") is triggered, causing an error because the file has already been renamed and can no longer be found under its previous name.

I'll take a look at the other similar tests

Reply:
That makes sense. Timers in tests should be avoided like the plague if possible.

Do we need to use an interval for operations like that? For example, would it work to use a `setTimeout()` and then reschedule the callback again once the timer fires?

Reply:
Hey @cjihrig, I agree, I hadn’t thought about replacing it with `setTimeout` 😬   
I’ll do it ASAP !

Reply:
@cjihrig, following another fix I was working on for a related flaky test, `test/parallel/test-runner-watch-mode-complex.mjs`, and following your suggestions, I removed the `setIntervals` in an attempt to simplify the test's readability while hopefully preventing flakiness